server/src/main/java/org/elasticsearch/action/support/RefCountingRunnable.java

```diff
@@ -103,7 +87,7 @@ public RefCountingRunnable(Runnable delegate) {
      */
     public Releasable acquire() {
         if (refCounted.tryIncRef()) {
-            return new AcquiredRef();
+            return Releasables.assertOnce(this);
```

Comment:
Probably deserves a comment tho:

```suggestion
            // All refs are considered equal so there's no real need to allocate a new object here, although note that this deviates
            // (subtly) with the docs for Closeable#close() which indicate that it should be idempotent. But only if assertions are
            // disabled, and if assertions are enabled then we are asserting that we never double-close these things anyway.
            return Releasables.assertOnce(this);
```